auth-jwt-service/src/main/java/io/stargate/auth/jwt/AuthzJwtService.java

```diff
@@ -0,0 +1,165 @@
+  private final Pattern tokenPattern = Pattern.compile("\\.");
+
+  /**
+   * Using the provided JWT and the claims it contains will perform pre-authorization where
+   * possible, executes the query provided, and then authorizes the response of the query.
+   *
+   * <p>{@inheritdoc}
+   */
+  @Override
+  public ResultSet authorizedDataRead(
+      Callable<ResultSet> action, String token, List<TypedKeyValue> typedKeyValues)
+      throws Exception {
+    JSONObject stargateClaims = extractClaimsFromJWT(token);
+
+    preCheckDataReadWrite(stargateClaims, typedKeyValues);
+
+    ResultSet result = action.call();
+
+    if (result == null) {
+      return null;
+    }
+
+    return result.withRowInspector(
+        row -> {
+          if (row == null) {
+            return true;
+          }
+
+          for (Column col : row.columns()) {
+            if (stargateClaims.has(STARGATE_PREFIX + col.name())) {
+
+              String stargateClaimValue;
+              try {
+                stargateClaimValue = stargateClaims.getString(STARGATE_PREFIX + col.name());
+              } catch (JSONException e) {
+                log.warn("Unable to get stargate claim for " + STARGATE_PREFIX + col.name());
+                return false;
+              }
+              String columnValue = row.getString(col.name());
+              if (!stargateClaimValue.equals(columnValue)) {
+                return false;
+              }
+            }
+          }
+          return true;
+        });
+  }
+
+  /**
+   * Using the provided JWT and the claims it contains will perform pre-authorization where possible
+   * and if successful executes the query provided.
+   *
+   * <p>{@inheritdoc}
+   */
+  @Override
+  public ResultSet authorizedDataWrite(
+      Callable<ResultSet> action, String token, List<TypedKeyValue> typedKeyValues)
+      throws Exception {
+    JSONObject stargateClaims = extractClaimsFromJWT(token);
+
+    preCheckDataReadWrite(stargateClaims, typedKeyValues);
+
+    // Just return the result. No value in doing a post check since we can't roll back anyway.
+    return action.call();
+  }
+
+  /**
+   * Authorization for schema resource access is not provided by JWTs so all authorization will be
+   * deferred to the underlying permissions assigned to the role the JWT maps to.
+   *
+   * <p>{@inheritdoc}
+   */
+  @Override
+  public void authorizeSchemaRead(String token, List<String> keyspaceNames, List<String> tableNames)
+      throws Exception {
+    // Cannot perform authorization with a JWT token so just return
+  }
+
+  /**
+   * Authorization for schema resource access is not provided by JWTs so all authorization will be
+   * deferred to the underlying permissions assigned to the role the JWT maps to.
+   *
+   * <p>{@inheritdoc}
+   */
+  @Override
+  public ResultSet authorizedSchemaWrite(
+      Callable<ResultSet> action, String token, String keyspace, String table) throws Exception {
+    return action.call();
+  }
```

Comment:
Would it be meaningful to align this method with `authorizeSchemaRead`, i.e. just do the authorization here and let the call invoke the action?